spec/tbd_tests_spec.rb

```diff
@@ -1566,3 +1566,414 @@
+          },
+          "additionalProperties": false,
+          "minProperties": 2,
+          "required": [
+            "id"
+          ]
+        },
+        "Space": {
+          "description": "Space default PSI-set (optional for OS)",
+          "type": "object",
+          "required": [
+            "id",
+            "psi"
+          ],
+          "properties": {
+            "id": {
+              "title": "e.g. OS space or E+ zone identifier",
+              "type": "string"
+            },
+            "psi": {
+              "title": "PSI-set identifier",
+              "type": "string"
+            }
+          },
+          "additionalProperties": false
+        },
+        "Story": {
+          "title": "Story default PSI-set (optional for OS)",
+          "type": "object",
+          "properties": {
+            "id": {
+              "title": "e.g. OS story identifier",
+              "type": "string"
+            },
+            "psi": {
+              "title": "PSI-set identifier",
+              "type": "string"
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "id",
+            "psi"
+          ]
+        },
+        "Unit": {
+          "title": "Building unit default PSI-set (optional for OS)",
+          "type": "object",
+          "properties": {
+            "psi": {
+              "title": "PSI-set identifier",
+              "type": "string"
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "psi"
+          ]
+        },
+        "Log": {
+          "title": "TBD log messages",
+          "type": "string"
+        }
+      }
+    }
+
+    tbd_o =
+    {
+      "schema": "https://github.com/rd2/tbd/blob/master/tbd.schema.json",
+      "description": "testing basic JSON validation",
+      "psis": [
+        {
+          "id": "good",
+          "parapet": 0.5
+        },
+        {
+          "id": "compliant",
+          "rimjoist": 0.3,
+          "parapet": 0.325,
+          "fenestration": 0.35,
+          "concave": 0.45,
+          "convex": 0.45,
+          "balcony": 0.5,
+          "party": 0.5,
+          "grade": 0.45
+        }
+      ],
+      "khis": [
+        {
+          "id": "column",
+          "point": 0.5
+        },
+        {
+          "id": "support",
+          "point": 0.5
+        }
+      ],
+      "edges": [
+        {
+          "psi": "compliant",
+          "type": "party",
+          "length": 2.5,
+          "surfaces": [
+            "front wall"
+          ]
+        }
+      ],
+      "surfaces": [
+        {
+          "id": "front wall",
+          "khis": [
+            {
+              "id": "column",
+              "count": 3
+            },
+            {
+              "id": "support",
+              "count": 4
+            }
+          ],
+          "psi": "good"
+        }
+      ],
+      "unit": [
+        {
+          "psi": "compliant"
```

Comment:
What is this example showing?  The unit is compliant, the surface is good, and the edge is compliant?  Would the other edges just not be listed here and they would adopt the values from their Space or Unit?  How would we know the type of edge for an edge that isn't listed here?  Is there a default edge type?

Reply:
So for starters, there's a mistake in the example. The first entry for "psis" should read :

```
{
  "id": "good",
  "parapet": 0.5,
  "party": 0.9
},
```

... my bad. Will correct in next commit. So the way one should read this input TBD JSON file:

- By default, all Topolys-detected edges shall be processed by TBD using the building unit default set "compliant".
- In addition, the surface "front wall" shall inherit 3x columns and 4x supports as _point_ thermal bridges (TBs). Wall with columns? [Flying buttresses](https://en.wikipedia.org/wiki/Flying_buttress)!
- Furthermore, all parapets and party wall edges of that same "front wall" shall not default to the "compliant" PSI-set, but rather the "good" set ... all other edge types (e.g. "fenestration") of the "front wall" shall continue to rely on the default "compliant" set.
- Exceptionally, a single party wall edge (2.5 m in height) is treated particularly efficiently, more evocative of the "compliant" PSI-set. So that specific edge shall not be processed by TBD using the "front wall"-specific "good" set, but revert back to the "compliant" set.

Is this a somewhat extreme case of customization? Yes! But something similar will undoubtedly be requested.